PocketKit/Sources/PocketKit/Home/HomeViewModel.swift

```diff
@@ -323,11 +330,48 @@ extension HomeViewModel {
                 contexts(for: cell, at: indexPath)
             )
         } else {
-            selectedReadableViewModel = RecommendationViewModel(
+            let viewModel = RecommendationViewModel(
                 recommendation: viewModel.recommendation,
                 source: source,
                 tracker: tracker.childTracker(hosting: .articleView.screen)
             )
+            selectedReadableType = .recommendation(viewModel)
+
+            tracker.track(
+                event: ContentOpenEvent(destination: .internal, trigger: .click),
+                contexts(for: cell, at: indexPath)
+            )
+        }
+    }
+
+    private func select(recentSave cell: HomeViewModel.Cell, at indexPath: IndexPath) {
+        guard case .recentSaves(let objectID) = cell,
+        let savedItem = bareItem(with: objectID) else {
+            return
+        }
+
+        tracker.track(
+            event: SnowplowEngagement(type: .general, value: nil),
+            contexts(for: cell, at: indexPath)
+        )
+
+        let item = savedItem.item
+        if let isArticle = item?.isArticle, isArticle == false
+            || item?.hasImage == .isImage
+            || item?.hasVideo == .isVideo {
+            presentedWebReaderURL = item?.bestURL
```

Comment:
Is this now the second place where this complex conditional logic lives? I think it could be pulled out into an extension `Item`.

```swift
extension Item {
  var shouldOpenInWebReader: Bool {
    // complex logic goes here
  }
```

What do you think?